psfws/gen_params.py

```diff
@@ -0,0 +1,371 @@
+        """
+        # pick out relevant wind data
+        raw_winds = self.get_raw_wind(pt)
+
+        # make a vector of heights where hufnagel will be valid
+        h_huf = np.linspace(self.h0 + 3, max(self.h_gfs), 100)
+        # interpolate wind data to those heights
+        speed_huf = self._interpolate_wind(raw_winds, h_huf)['speed'].flatten()
+        # calculate hufnagel cn2 for those
+        cn2_huf = utils.hufnagel(h_huf, speed_huf)
+
+        # draw model of ground turbulence
+        cn2_gl, h_gl = self._draw_ground_cn2()
+
+        cn2_complete = np.hstack([cn2_gl, cn2_huf])
+        h_complete = np.hstack([h_gl, h_huf])
+
+        # return stacked hufnagel and ground layer profiles/altitudes
+        return cn2_complete, h_complete
+
+    def get_cn2_all(self):
+        """Get array of Cn2 values for all data available."""
+        cn2_list = []
+        for i in range(self.N):
+            cn2, h = self.get_cn2(i)
+            cn2_list.append(cn2)
+        return np.array(cn2_list), h
+
+    def _get_auto_layers(self):
+        """Return layer altitudes according to max wind speed & turbulence."""
+        # make an array of heights for interpolation
+        h_interp = np.linspace(self.h0 + self.h_dome, max(self.h_gfs), 500)
+
+        # interpolate the median speeds from GFS to find height of max
+        all_speeds = [i for i in self.gfs_winds['speed'].values]
+        h_maxspd = utils.find_max_median(all_speeds, self.h_gfs,
+                                         h_interp, self.h0)
+
+        # interpolate the median cn2 to find height of max
+        all_cn2, h_cn2 = self.get_cn2_all()
+        h_maxcn2 = utils.find_max_median(all_cn2, h_cn2, h_interp, self.h0)
+
+        # sort the heights of max speed and max turbulence
+        h3, h4 = np.sort([h_maxspd, h_maxcn2])
+
+        # raise the lowest layer slightly off of the ground
+        lowest = self.h0 + 0.250
+
+        h2 = np.mean([lowest, h3])
+        h5 = np.mean([h4, 18])
+
+        return [lowest, h2, h3, h4, h5, 18]
+
+    def _integrate_cn2(self, cn2, h, layers='auto'):
+        """Calculate turbulence integral of given Cn2 at given layers.
+
+        Parameters
+        ----------
+        cn2 : array
+            cn2 values
+        h : array
+            heights of cn2 values
+        layers : str or list or array
+            Centers of integration regions set by layers keyword; either
+            list/array of values, or 'auto' for them to be automatically
+            calculated based on wind and turbulence maximums (default: 'auto')
+
+        Returns
+        -------
+        turbulence integral values J, in m^(1/3)
+        integration bin edges, in km
+        integration bin centers (aka layers if not 'auto'), in km
+
+        """
+        maxh = max(self.h_gfs)
+
+        # define bins according to layers argument
+        if layers == 'auto':
+            bin_centers = self._get_auto_layers()
+        else:
```

Comment:
you could add a quick check that the layers manually provided are not completely nonsensical (eg given in meters)